### PR TITLE
Fix timm fp16 accuracy for new models

### LINKED ISSUE #167608: New timm models in benchmarks failing accuracy
When these models were added in https://github.com/pytorch/pytorch/pull/165381 , they were already failing accuracy (see hud results):

convnextv2_nano.fcmae_ft_in22k_in1k FAIL: accuracy=fail_accuracy, expected=pass
Error: 1 models have accuracy status regressed:
vit_base_patch14_dinov2.lvd142m

We should investigate and fix them, I assume we added them bc they were important. @yiming0416 

cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #185038

The new TIMM models were running benchmark training accuracy with the default capturable Adam path. In fp16, eager Adam produces NaNs for these models, so the benchmark compares Inductor against an unstable reference and reports fail_accuracy.

The benchmark already routes models with unstable eager Adam training accuracy through SGD. Add the two new TIMM models to that existing policy, and update the tested CUDA static and dynamic Inductor expected baselines for vit_base_patch14_dinov2.lvd142m to pass. convnextv2_nano.fcmae_ft_in22k_in1k was already expected to pass on CUDA; ROCm static baselines are left unchanged because this change was not validated on ROCm.

Fixes #167608

Generated by my agent

Test Plan:

- wrapped timm_models.py --accuracy --training --only=convnextv2_nano.fcmae_ft_in22k_in1k --backend=inductor --float16: pass

- wrapped timm_models.py --accuracy --training --only=vit_base_patch14_dinov2.lvd142m --backend=inductor --float16: pass

- wrapped timm_models.py --accuracy --training --only=vit_base_patch14_dinov2.lvd142m --backend=inductor --float16 --dynamic-shapes: pass

- python benchmarks/dynamo/check_accuracy.py --actual accuracy_inductor.csv --expected benchmarks/dynamo/ci_expected_accuracy/inductor_timm_training.csv: PASS

- python benchmarks/dynamo/check_accuracy.py --actual accuracy_inductor.csv --expected benchmarks/dynamo/ci_expected_accuracy/dynamic_inductor_timm_training.csv: PASS

- lintrunner -a: ok

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I updated the PR for the inductor_timm CI failures. The previous run showed convnextv2_nano.fcmae_ft_in22k_in1k improving to 6 graph breaks and vit_base_patch14_dinov2.lvd142m OOMing while generating the fp64 golden reference before falling back to cosine fail_accuracy; this revision updates the convnext expected graph-break baseline and adds a CI-only smaller accuracy batch for the ViT model so the fp64 reference path can run.

**Comment by jansel** [comment]
My agent says I updated the PR for the latest inductor_timm CI failure. The failing shards passed accuracy but failed check_graph_breaks because vit_base_patch14_dinov2.lvd142m improved to graph_breaks=6 while the expected baseline was 7; this revision updates the CUDA Inductor TIMM training baselines and scopes the SGD and CI batch-size workarounds to CUDA Inductor so eager/AOT/CPU/ROCm baselines are not unintentionally affected.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I repaired the stale ghstack PR head and resubmitted the fix so CI is now running on the updated commit. This revision also tightens the new workarounds to the failing CUDA Inductor TIMM training accuracy surface only: SGD is accuracy-only, and the smaller ci_accuracy batch size is training-only.

**Comment by jansel** [comment]
My agent says I rebased the PR onto current main to pick up the docs preview upload revert. The prior docs failures were from the old workflow path failing at `aws s3 sync` with `aws: command not found`; CI is now rerunning on the rebased head.

**Review by karthickai — APPROVED** [review]

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
@claude are those CI failures caused by this PR?
